File: May Leetcode Challenge/Python/581. Shortest Unsorted Continuous Subarray.py
class Solution:
    def findUnsortedSubarray(self, nums: List[int]) -> int:
        n = len(nums)
        first = nums[0]
        last = nums[-1]
        start = 0
        end = 0
        for i in range(1,n):
            if nums[i]<first:
                end = i
            else:
                first = nums[i]
            if last<nums[~i]:
                start = ~i
            else:
                last = nums[~i]
        # print(first, end)
        if end!=0:
            return end - (n+start) + 1
        return 0

File: May Leetcode Challenge/581. Shortest Unsorted Continuous Subarray.cpp
class Solution {
public:
    int findUnsortedSubarray(vector<int>& nums) {
        int n=nums.size();
        int l=0,r=n-1,maxn=0,minn=n-1;
        for(int i=0;i<n;i++)
        {
            if(nums[i]>nums[maxn])
            {
                maxn=i;
            }
            else if(nums[maxn]>nums[i])
            {
                l=i;
            }
            if(nums[n-i-1]<nums[minn])
            {
                minn=n-i-1;
            }
            else if(nums[n-i-1]>nums[minn])
            {
                r=n-i-1;
            }
        }
       if(r>=l)
       {
           return 0;
       }
        return l-r+1;
    }
};
